src/server/db/migrations/2020-11-13_0.3.0_to_0.4.0/migrate_0.3.0_to_0.4.0.js

```diff
@@ -0,0 +1,414 @@
+        auth: options.parsedCouchDbUrl.auth,
+        method: "POST",
+        headers: {
+            "Accept": "application/json",
+            "Content-Length": 0, // this value is filled in at runtime
+            "Content-Type": "application/json"
+        }
+    };
+    fluid.log("COUCHDB_URL: '" +
+        options.parsedCouchDbUrl.protocol + "//" +
+        options.parsedCouchDbUrl.hostname + ":" +
+        options.parsedCouchDbUrl.port +
+        options.parsedCouchDbUrl.pathname + "'"
+    );
+    return options;
+};
+
+/**
+ * Create the step that retrieves stories from the database
+ *
+ * @param {Object} options - options for the query
+ *
+ * @return {Promise} - A promise that resolves retrieving the stories.
+ */
+sjrk.storyUpdate.retrieveStories = function (options) {
+    var details = {
+        requestUrl: options.storiesUrl,
+        requestErrMsg: "Error retrieving stories from the database: ",
+        responseDataHandler: sjrk.storyUpdate.updateStoriesData,
+        responseErrMsg: "Error retrieving stories from database: "
+    };
+    return gpii.dbRequest.configureStep(details, options);
+};
+
+/**
+ * Given all the documents from the database, update their data according to the new data model
+ *
+ * @param {String} responseString - the response from the request to get all documents
+ * @param {Object} options - Where to store the to-be-updated documents
+ *
+ * @return {Array} - The updated documents in the new data structure with the new schema version.
+ */
+sjrk.storyUpdate.updateStoriesData = function (responseString, options) {
+    var allDocs = JSON.parse(responseString);
+    var updatedDocs = [];
+    options.totalNumOfDocs = allDocs.total_rows;
+
+    if (allDocs.rows) {
+        fluid.each(allDocs.rows, function (aRow) {
+            var aDoc = aRow.doc;
+            // To filter out the "_design/views" doc that doesn't have the "schemaVersion" field
+            if (aDoc.type === "story") {
+                // Make the required updates, adding missing fields
+                // Please consult the list at the top of this file for details on each item
+
+                if (!aDoc.schemaVersion || aDoc.schemaVersion < sjrk.storyUpdate.newSchemaVersion) {
+
+                    aDoc.schemaVersion = sjrk.storyUpdate.newSchemaVersion; // 2.a
+
+                    aDoc.value.id = aDoc.value.id || aDoc._id; // 2.b
+                    aDoc.value.published = aDoc.value.published || true; // 2.c
+                    aDoc.value.timestampCreated = aDoc.value.timestampCreated || new Date().toDateString(); // 2.d
+                    aDoc.value.timestampPublished = aDoc.value.timestampPublished || new Date().toDateString(); // 2.e
+
+                    var blocks = fluid.copy(aDoc.value.content);
+
+                    fluid.each(blocks, function (block, i) {
+                        block.order = block.order || i; // 2.f
+                        block.firstInOrder = block.firstInOrder || i === 0 ? true : false; // 2.g
```

Comment:
For these two lines, recall that ternary booleans are always redundant

Reply:
An old habit from my C# days :) Thanks for pointing that one out!